src/go/k8s/pkg/resources/statefulset_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package resources_test
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	res "github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestEnsure(t *testing.T) {
+	cluster := pandaCluster()
+	stsResource := stsFromCluster(cluster)
+
+	var newReplicas int32 = 3333
+
+	replicasUpdatedCluster := cluster.DeepCopy()
+	replicasUpdatedCluster.Spec.Replicas = &newReplicas
+	replicasUpdatedSts := stsFromCluster(cluster).DeepCopy()
+	replicasUpdatedSts.Spec.Replicas = &newReplicas
+
+	newResources := corev1.ResourceList{
+		corev1.ResourceCPU:	resource.MustParse("1111"),
+		corev1.ResourceMemory:	resource.MustParse("2222Gi"),
+	}
+	resourcesUpdatedCluster := cluster.DeepCopy()
+	resourcesUpdatedCluster.Spec.Resources.Requests = newResources
+	resourcesUpdatedSts := stsFromCluster(cluster).DeepCopy()
+	resourcesUpdatedSts.Spec.Template.Spec.Containers[0].Resources.Requests = newResources
+
+	var tests = []struct {
+		name		string
+		existingObject	client.Object
+		pandaCluster	*redpandav1alpha1.Cluster
+		expectedObject	*v1.StatefulSet
+	}{
+		{"no existing", nil, cluster, stsResource},
+		{"update replicas", stsResource, replicasUpdatedCluster, replicasUpdatedSts},
+		{"update resources", stsResource, resourcesUpdatedCluster, resourcesUpdatedSts},
+	}
+
+	for _, tt := range tests {
+		c := fake.NewClientBuilder().Build()
+
+		err := redpandav1alpha1.AddToScheme(scheme.Scheme)
+		if err != nil {
+			t.Errorf("%s: expecting no error, got %v", tt.name, err)
+		}
+
+		if tt.existingObject != nil {
+			tt.existingObject.SetResourceVersion("")
+			err = c.Create(context.Background(), tt.existingObject)
```

Comment:
nit: it's inconsistent with line 78. Should we cuddle an `if` statement checking an error variable with the line that assigned that error variable?

Let's disable wsl, but stick to one convention.

Reply:
👍 I personally here had no newline (before golangcilint forced me to create one :))